app/presenters/special_announcement_presenter.rb

```diff
@@ -0,0 +1,36 @@
+class SpecialAnnouncementPresenter
+  attr_reader :content_item
+
+  def initialize(content_item)
+    @content_item = content_item
+  end
+
+  def payload
+    @payload ||= creative_work.merge(special_announcement_properties)
+  end
+
+private
+
+  def special_announcement_properties
+    schema_specific_links = content_item.dig("details", "special_announcement_schema") || {}
+
+    {
+      "@type" => "SpecialAnnouncement",
+      "category" => schema_specific_links["category"],
```

Comment:
Hey if any of these variables do not exist, does it fail gracefully?

Reply:
Yep, the value gets set to `nil` and the `compact` at the end of the hash drops it from the schema.

Reply:
You can check this in the preview app because we don't have a URL for `gettingTestedInfo` yet

Reply:
Will these null values cause any issues?

```javascript
{
  "@context": "http://schema.org",
  "@type": "SpecialAnnouncement",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://www.gov.uk/coronavirus"
  },
  "headline": "Coronavirus (COVID-19): what you need to do",
  "datePublished": "2020-03-20T14:29:42.000+00:00",
  "dateModified": "2020-03-30T16:57:50.000+00:00",
  "description": "Find out about the government response to coronavirus (COVID-19) and what you need to do.",
  "publisher": {
    "@type": "Organization",
    "name": "GOV.UK",
    "url": "https://www.gov.uk",
    "logo": {
      "@type": "ImageObject",
      "url": null
    }
  },
  "image": null,
  "category": "https://www.wikidata.org/wiki/Q81068910",
  "datePosted": "2020-03-30T16:57:50.000+00:00",
  "diseasePreventionInfo": "https://www.gov.uk/coronavirus",
  "diseaseSpreadStatistics": "https://www.gov.uk/government/publications/covid-19-track-coronavirus-cases",
  "newsUpdatesAndGuidelines": "https://www.gov.uk/coronavirus",
  "publicTransportClosuresInfo": "https://www.gov.uk/guidance/coronavirus-covid-19-uk-transport-and-travel-advice",
  "quarantineGuidelines": "https://www.gov.uk/government/publications/coronavirus-outbreak-faqs-what-you-can-and-cant-do/coronavirus-outbreak-faqs-what-you-can-and-cant-do",
  "schoolClosuresInfo": "https://www.gov.uk/check-school-closure",
  "travelBans": "https://www.gov.uk/guidance/travel-advice-novel-coronavirus"
}
```

Reply:
I tweaked the test to prove this :-)